src/hooks/connectWeb3/useActiveWeb3React.tsx

```diff
@@ -132,32 +134,31 @@ function toWeb3Connection<T extends Connector>([connector, hooks]: [T, Web3React
   return [connector, hooks]
 }
 
-function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint?: string | JsonRpcProvider) {
-  // TODO(kristiehuang): implement RPC URL fallback, then jsonRpcEndpoint can be optional
-
+function getWalletConnectConnection(useDefault: boolean, jsonRpcEndpoint: string | JsonRpcProvider) {
   // WalletConnect relies on Buffer, so it must be polyfilled.
   if (!('Buffer' in window)) {
     window.Buffer = Buffer
   }
 
-  let rpcUrl: string
+  // FIXME(kristiehuang): we don't know what the props.jsonRpcEndpoint chain is; assume mainnet for WC instantiation
+  let mainnetRpcUrl: string
   if (JsonRpcProvider.isProvider(jsonRpcEndpoint)) {
-    rpcUrl = jsonRpcEndpoint.connection.url
+    mainnetRpcUrl = jsonRpcEndpoint.connection.url
   } else {
-    rpcUrl = jsonRpcEndpoint ?? '' // TODO(kristiehuang): use fallback RPC URL
+    mainnetRpcUrl = jsonRpcEndpoint
   }
-
   return toWeb3Connection(
     initializeConnector<WalletConnect>(
       (actions) =>
         new WalletConnect({
           actions,
           options: {
             rpc: {
-              1: [rpcUrl].filter((url) => url !== undefined && url !== ''),
+              [SupportedChainId.MAINNET]: mainnetRpcUrl,
```

Comment:
You may want to reference this code in the earlier TODO as something to change when multiple jsonRpcEndpoints (for multiple chains) are supported).